samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -233,6 +241,10 @@ public ClusterBasedJobCoordinator(MetricsRegistryMap metrics, MetadataStore meta
             new ApplicationConfig(config));
     this.containerPlacementRequestAllocatorThread =
         new Thread(containerPlacementRequestAllocator, "Samza-" + ContainerPlacementRequestAllocator.class.getSimpleName());
+
+    // build DrainMonitor
+    this.drainMonitor = new DrainMonitor(metadataStore, config);
```

Comment:
As discussed offline, we don't need to listen to the drain message in JC. So let's not create DrainMonitor here.

Reply:
Done.